File: src/main/java/com/dalolorn/sr2modmanager/adapter/SingleFinder.java
package com.dalolorn.sr2modmanager.adapter;

import org.jetbrains.annotations.Nullable;

import java.nio.file.Path;

public class SingleFinder extends Finder<Path> {
	@Nullable
	private Path result;

	public SingleFinder(String pattern) {
		super(pattern);
	}

	// Compares the glob pattern against
	// the file or directory name.
	@Nullable
	protected Path find(Path file) {
		Path name = file.getFileName();
		if (name != null && matcher.matches(name)) {
			result = file;
			return file;
		}
		return null;
	}

	// Returns the result.
	@Nullable
	public Path getResult() {
		return result;
	}
}

File: src/main/java/io/github/openstarruler/launchpad/adapter/SingleFinder.kt
package io.github.openstarruler.launchpad.adapter

import java.nio.file.Path

class SingleFinder(pattern: String) : Finder<Path?>(pattern) {
    // Returns the result.
    override var result: Path? = null
        private set

    // Compares the glob pattern against
    // the file or directory name.
    override fun find(file: Path): Path? {
        val name = file.fileName
        if (name != null && matcher.matches(name)) {
            result = file
            return file
        }
        return null
    }
}